app/middleware/preview-api.js

```diff
@@ -0,0 +1,35 @@
+const contentStore = require('../../lib/content-store');
+const parseUrl = require('parseurl');
+const get = require('lodash/get');
+const crypto = require('crypto');
+const timingSafeCompare = require('tsscmp');
+
+/* eslint-disable no-param-reassign */
+module.exports = (req, res, next) => {
+  const pathName = get(parseUrl.original(req), 'pathname', '');
+  const [, givenSignature, pageId, revisionId] = pathName.match(/\/preview\/(.*=)\/(\d+)\/(\d+)/);
```

Comment:
@nickcolley JS's `match` method returns an array of values. The first value in the array is the entire match followed by parentheses-captured matches. In this case we're only interested in the captured matches so we ignore the first value by skipping it in the destructuring list.

[Ignoring values in destructuring assignment](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Ignoring_some_returned_values).

Although we should protect it against a possible `null` value.